caddy.go

```diff
@@ -38,6 +39,15 @@ import (
 	"go.uber.org/zap"
 )
 
+func init() {
+	RegisterType("", []reflect.Type{
```

Comment:
Good point! How about this?

```go
func RegisterType(namespace string, vals []interface{}) {
	var types []reflect.Type
	for _, v := range vals {
		reflect.TypeOf(v).Elem()
	}
	if _, ok := namespaceTypes[namespace]; ok {
		panic("namespace is already registered")
	}
	namespaceTypes[namespace] = types
}
```

To be used as such:
```go
caddy.RegisterType("http.authentication.hashes", []interface{}{
	(*Comparer)(nil),
})
```

I did consider variadic function (i.e. `func RegisterType(namespace string, vals []interface{})`) just now, but it'll silently fail if a user passes a slice, i.e.  `RegisterType("http.authentication.hashes", []interface{}{(*Comparer)(nil)})`. The slice version (i.e. `[]interface{}`) is more intentional.

Reply:
Yeah, I like the proposed change there. And I agree that a variadic function isn't necessary.

I am still on the fence as to whether this is a useful/needed addition to Caddy, though; still, I want to thoroughly review it since you took the time to implement it.